dts/bindings/sensor/ti,tmp108.yaml

```diff
@@ -16,3 +17,8 @@ properties:
       description: |
         Identifies the ALERT signal, which is active-low open drain when
         produced by the sensor.
+  
+    variant-as621x:
```

Comment:
I think the right way to solve this case is to make the driver support two compatibles, factorizing common code. cc: @mbolivar-nordic

Reply:
Agreed

Reply:
thanks, @gmarull and @mbolivar-nordic.  Sorry just getting back from a long break on this, so that's what we attempted to do.  1 register set is included if there is an AMS621 included in the device tree otherwise the register set from the TMP108 is included. Just wanted to make sure the approach is sound before continuing on with it.

Reply:
@gmarull @mbolivar-nordic done, please take a look when you have a few minutes.

Reply:
Did you forget to push, @catch-twenty-two ? I see no changes.

Reply:
Hi @mbolivar-nordic , I did make a bunch of changes. The drivers share a common .h file and the few registers definitions that are different are included depending on which driver is chosen in the KConfig example (line 22):
```
#if CONFIG_AS621x
#include "as621x_defs.h"
#else
#include "tmp108_defs.h"
#endif
```

Reply:
> Did you have something else in mind?

Yes. Take another look at @gmarull's advice (emphasis mine):

> I think the right way to solve this case is to **make the driver support two compatibles**, factorizing common code. 

I.e., you need two different devicetree bindings, one for each compatible. The driver should support **both** compatibles, and importantly both **at the same time**.

Reply:
@mbolivar-nordic Thanks for your help, I have reworked everything so this driver now supports multiple instances of both the tmp108 and as6212 on the same board, if my approach looks ok Ill go ahead and clean everything up.

Reply:
@gmarull fyi

Reply:
>  I have reworked everything so this driver now supports multiple instances of both the tmp108 and as6212 on the same board, if my approach looks ok Ill go ahead and clean everything up.

Did you forget to push something? I still see the problematic `variant-as621x` property.

Regardless of whether you create two drivers or create one driver that supports two compatibles, you still need two devicetree bindings.

Remember: a compatible property says what type of hardware the node represents. Having compatible be "ti,tmp108", but then adding a property that says "just kidding, this is actually an as621x", is not the right thing to do. Two parts --> two compatibles.

Reply:
Ah okay, thanks for the information, I didn't realize that, will make the change